### ISSUE TITLE
[a11y]: 4.1.2 Name, Role, Value Violation from <FilterableMultiSelect /> aria-describedby label

### ISSUE THREAD

**Issue by matthewpagacik12** [carbon-design-system/carbon#19644]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.78.0

### React version

18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The issue we are experiencing is a 4.1.2 Name, Role, Value violation on our input element. This issue is happening because we are using the `helperText` prop on the `<FilterableMultiSelect />` component and that text when it is not present on the screen (it is not present for us when our input is invalid) hits the `helperId` and that will set the `helperId` to `filterablemultiselect-helper-text-${filterableMultiSelectInstanceId}`. This then causes issues because the `aria-describedby` is set to the `helperId` when `helperText` is `true`. Then when the `helperText` `<div>` disappears on our screen, the error message appears but the `filterablemultiselect-helper-text-${filterableMultiSelectInstanceId}` is not equal to the `id` of the `helperText` `<div>`

![Image](https://github.com/user-attachments/assets/a76097b7-1fb6-400b-a08a-e04802c214d6)

![Image](https://github.com/user-attachments/assets/4538fc98-0758-4c20-ba12-a7a8631a47cf)

![Image](https://github.com/user-attachments/assets/7a61fe0d-49cd-4470-bfa5-41ac143dd43f)
![Image](https://github.com/user-attachments/assets/3026eeda-275d-43fe-bf0d-8b5a4353ca98)
![Image](https://github.com/user-attachments/assets/672461de-b9b1-4839-93fb-75f33ddf5f43)
![Image](https://github.com/user-attachments/assets/dd550862-1141-48c0-b304-3f7bf6820c65)

### WCAG 2.1 Violation

https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

### Reproduction/example

https://ecasr6c4--run.stackblitz.io/

### Steps to reproduce

1.) Go to the link above: https://stackblitz.com/run?file=src%2FApp.jsx

2.) Open up the IBM Equal Access Accessibility Checker chrome tool

3.) Make sure the input is invalid and look for the 4.1.2 accessibility violation.

![Image](https://github.com/user-attachments/assets/1d539f0d-95c9-409c-a7a4-7d2003e8f731)

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by adamalston** [comment]
Similar to https://github.com/carbon-design-system/carbon/issues/19472.

**Comment by tay1orjones** [comment]
@heloiselui do you think your work in #19472 could be applied to this as well?

**Comment by heloiselui** [comment]
> [@heloiselui](https://github.com/heloiselui) do you think your work in [#19472](https://github.com/carbon-design-system/carbon/issues/19472) could be applied to this as well?

~Not 100% sure, but let me check that! Thanks @tay1orjones and @adamalston~ I confirm that the issue is unrelated, so the [fix](https://github.com/carbon-design-system/carbon/pull/19922) for #19472 will not resolve this a11y problem.

**Comment by fbarroso24** [comment]
Hi @tay1orjones @heloiselui, I was wondering if there were any updates on this.  

This is a high priority for us as it is a 4.1.2 violation & that is part of the A11Y requirements that are required for FedRAMP.

**Comment by tay1orjones** [comment]
@fbarroso24 thanks for letting us know this relates to FedRAMP. I've got a PR up.

**Comment by fbarroso24** [comment]
> [@fbarroso24](https://github.com/fbarroso24) thanks for letting us know this relates to FedRAMP. I've got a PR up.

Thanks @tay1orjones! Much appreciated.
